proxy/envoy/discovery.go

```diff
@@ -302,6 +321,37 @@ func (ds *DiscoveryService) clearCache() {
 	ds.rdsCache.clear()
 }
 
+// ListAllEndpoints responds with all Services and is restricted to a single service key
+func (ds *DiscoveryService) ListAllEndpoints(request *restful.Request, response *restful.Response) {
+
+	var servicesArray []*service
+	for _, svc := range ds.services.Services() {
+		for _, port := range svc.Ports {
+
+			var hostArray []*host
+			for _, ep := range ds.services.Instances(svc.Hostname, []string{port.Name}, []model.Tags{}) {
```

Comment:
Preallocating the slice is not strictly necessary since this isn't in the fast path. That said, the original declaration is not 100% correct since hostArray is eventually encoded as a JSON object (see https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices). 

Any of the following variants should work, though I have a preference for the first since it's the least amount of code.

```golang
hosts := []*host
hosts := make([]*host)
hosts := make([]*host, 0, len(instances))
```